src/Sylius/Bundle/AddressingBundle/Repository/ZoneRepository.php

```diff
@@ -0,0 +1,126 @@
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Addressing\Repository\ZoneRepositoryInterface;
+
+/**
+ * @implements ZoneRepositoryInterface<ZoneInterface>
+ */
+class ZoneRepository extends EntityRepository implements ZoneRepositoryInterface
+{
+    public function findOneByAddress(AddressInterface $address, ?string $scope = null): ?ZoneInterface
+    {
+        $query = $this->createFindByAddressQueryBuilder($address, $scope);
+
+        $query
+            ->addSelect('(CASE
+                    WHEN z.type = \'province\' THEN 1
+                    WHEN z.type = \'country\' THEN 2
+                    WHEN z.type = \'zone\' THEN 3
+                    ELSE 4
+                END) AS HIDDEN sort_order')
+            ->orderBy('sort_order', 'ASC')
+            ->setMaxResults(1)
+        ;
+
+        return $query->getQuery()->getOneOrNullResult();
+    }
+
+    /** @return ZoneInterface[] */
+    public function findAllByAddress(AddressInterface $address, ?string $scope = null): array
+    {
+        return $this->createFindByAddressQueryBuilder($address, $scope)->getQuery()->getResult();
+    }
+
+    public function createFindByAddressQueryBuilder(AddressInterface $address, ?string $scope = null): QueryBuilder
+    {
+        $query = $this->createQueryBuilder('z')
+            ->select('z', 'm')
+            ->leftJoin('z.members', 'm')
+        ;
+
+        if (null !== $scope) {
+            $query
+                ->andWhere($query->expr()->in('z.scope', ':scopes'))
+                ->setParameter('scopes', [$scope, Scope::ALL])
+            ;
+        }
+
+        $orConditions = [];
+
+        if ($address->getCountryCode() !== null) {
+            $orConditions[] = $query->expr()->andX(
+                $query->expr()->eq('z.type', ':country'),
+                $query->expr()->eq('m.code', ':countryCode'),
+            );
+
+            $query->setParameter('country', ZoneInterface::TYPE_COUNTRY);
+            $query->setParameter('countryCode', $address->getCountryCode());
+        }
+
+        if ($address->getProvinceCode() !== null) {
+            $orConditions[] = $query->expr()->andX(
+                $query->expr()->eq('z.type', ':province'),
+                $query->expr()->eq('m.code', ':provinceCode'),
+            );
+
+            $query->setParameter('province', ZoneInterface::TYPE_PROVINCE);
+            $query->setParameter('provinceCode', $address->getProvinceCode());
+        }
+
+        $query->andWhere($query->expr()->orX(...$orConditions));
+
+        return $query;
+    }
+
+    /**
+     * @param array<ZoneInterface> $members
+     *
+     * @return array<ZoneInterface>
+     */
+    public function findZonesByMembers(array $members, ?string $scope = null): array
+    {
+        $zonesCodes = array_map(
+            fn (ZoneInterface $zone): string => $zone->getCode(),
+            $members,
+        );
+
+        $query = $this->createQueryBuilder('z')
```

Comment:
In what I've seen so far in sylius, repositories don't change default alias and uses `o`.

```patch
--- $this->createQueryBuilder('z')
+++ $this->createQueryBuilder('o')
```